data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -220,17 +220,19 @@ public void completePartition(final String partitionKey) {
             throw e;
         }
 
-        partitionManager.removeActivePartition();
+        if (!fromAcknowledgmentsCallback) {
+            partitionManager.removeActivePartition();
+        }
 
         LOG.info("Partition key {} was completed by owner {}.", partitionKey, ownerId);
         partitionsCompletedCounter.increment();
     }
 
     @Override
-    public void closePartition(final String partitionKey, final Duration reopenAfter, final int maxClosedCount) {
+    public void closePartition(final String partitionKey, final Duration reopenAfter, final int maxClosedCount, final Boolean fromAcknowledgmentsCallback) {
         validateIsInitialized();
 
-        final SourcePartitionStoreItem itemToUpdate = validateAndGetSourcePartitionStoreItem(partitionKey, "close");
+        final SourcePartitionStoreItem itemToUpdate = fromAcknowledgmentsCallback ? getSourcePartitionStoreItem(partitionKey, "close") : validateAndGetSourcePartitionStoreItem(partitionKey, "close");
```

Comment:
This logic is very similar to what you have above in another method. It would be best to move this into a private method so that future maintainers/developers don't miss the other location when modifying the code.

Reply:
Can you add a comment on why validation is not needed if this is called from acknowledgement callback?